File: app/src/main/java/com/dolphkon/wanandroidclient/net/DataEncryptInterceptor.java
package com.dolphkon.wanandroidclient.net;

import com.dolphkon.httplib.utils.LogUtil;

import java.io.IOException;
import java.net.URLDecoder;

import okhttp3.FormBody;
import okhttp3.Interceptor;
import okhttp3.MediaType;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;
import okhttp3.ResponseBody;
import okio.Buffer;

public class DataEncryptInterceptor implements Interceptor {
    @Override
    public Response intercept(Chain chain) throws IOException {
        //请求
        Request request = chain.request();
        RequestBody oldRequestBody = request.body();

        String oldBodyStr = "";
        if (oldRequestBody != null) {
            Buffer requestBuffer = new Buffer();
            oldRequestBody.writeTo(requestBuffer);
            oldBodyStr = requestBuffer.readUtf8();
            requestBuffer.close();
        }
        MediaType mediaType = MediaType.parse("application/json;charset=UTF-8");
        String method = request.method();
        if ("POST".equals(method)) {
            StringBuilder sb = new StringBuilder();
            if (request.body() instanceof FormBody) {
                FormBody body = (FormBody) request.body();
                for (int i = 0; i < body.size(); i++) {
                    if(body.encodedName(i)!=null&&body.encodedName(i).equals("common")){
                        sb.append("\""+body.encodedName(i)+"\"" + ":" + body.encodedValue(i) + ",");
                    }else{
                        sb.append("\""+body.encodedName(i)+"\"" + ":\"" + body.encodedValue(i) + "\",");
                    }
                }
                sb.delete(sb.length() - 1, sb.length());
                LogUtil.e("请求：\n" + request.url() + "\n{" + URLDecoder.decode(sb.toString(), "utf-8") + "}");
            } else {
                LogUtil.e("请求：\n" + request.url()+","+oldBodyStr);
            }
        } else {
            LogUtil.e(String.format("请求：\n%s\n%s",
                    request.url(), request.headers()));
        }

        //响应
        Response response = chain.proceed(request);
        ResponseBody responseBody = response.peekBody(1024 * 1024);

        LogUtil.e("响应：\n" + responseBody.string() + "");
        return response;
    }
}

File: app/src/main/java/com/dolphkon/httpClient/net/DataEncryptInterceptor.kt
package com.dolphkon.httpClient.net

import com.dolphkon.httplib.utils.LogUtil
import okhttp3.FormBody
import okhttp3.Interceptor
import okhttp3.Response
import okio.Buffer
import java.io.IOException
import java.net.URLDecoder

class DataEncryptInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response { //请求
        val request = chain.request()
        val oldRequestBody = request.body()
        var oldBodyStr = ""
        if (oldRequestBody != null) {
            val requestBuffer = Buffer()
            oldRequestBody.writeTo(requestBuffer)
            oldBodyStr = requestBuffer.readUtf8()
            requestBuffer.close()
        }
        val method = request.method()
        if ("POST" == method) {
            val sb = StringBuilder()
            if (request.body() is FormBody) {
                val body = request.body() as FormBody?
                for (i in 0 until body!!.size()) {
                    if (body.encodedName(i) != null && body.encodedName(i) == "common") {
                        sb.append("\"" + body.encodedName(i) + "\"" + ":" + body.encodedValue(i) + ",")
                    } else {
                        sb.append("\"" + body.encodedName(i) + "\"" + ":\"" + body.encodedValue(i) + "\",")
                    }
                }
                sb.delete(sb.length - 1, sb.length)
                LogUtil.e("请求：\n" + request.url() + "\n{" + URLDecoder.decode(sb.toString(), "utf-8") + "}")
            } else {
                LogUtil.e("请求：\n" + request.url() + "," + oldBodyStr)
            }
        } else {
            LogUtil.e(String.format("请求：\n%s\n%s",
                    request.url(), request.headers()))
        }
        //响应
        val response = chain.proceed(request)
        val responseBody = response.peekBody(1024 * 1024.toLong())
        LogUtil.e("响应：\n" + responseBody.string() + "")
        return response
    }
}